### ISSUE TITLE
Typescript Compile Error: Cannot find namespace JSX

### ISSUE THREAD

**Issue by darylteo** [NightCatSama/vue-slider-component#321]
#### Describe the bug

Tested Versions: 3.0.12, 3.0.10
Typescript Version: 3.3.3333

```

vue-slider-component/typings/vue-slider.d.ts(108,15):

TS2503: Cannot find namespace 'JSX'.

```

Reverting to 2.8.14 resolves issue.

**Comment by NightCatSama** [comment]
@darylteo Which method did you import and use?

**Comment by sleewoo** [comment]
Encountering this as well.

Importing it globally:

<img width="504" alt="Screen Shot 2019-03-14 at 10 15 50" src="https://user-images.githubusercontent.com/9264860/54341346-5cc96880-4642-11e9-8122-5e1103bb3832.png">

And here is the output:

<img width="1021" alt="Screen Shot 2019-03-14 at 10 14 28" src="https://user-images.githubusercontent.com/9264860/54341363-6c48b180-4642-11e9-90c4-ee0b2c45a64a.png">

**Comment by NightCatSama** [comment]
The project I generated using `@vue/cli` does not have this problem. What is your `tsconfig.json`?

This is `tsconfig.json` generated by `@vue/cli`.
```json
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "strict": true,
    "jsx": "preserve",
    "importHelpers": true,
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "sourceMap": true,
    "baseUrl": ".",
    "types": [
      "webpack-env"
    ],
    "paths": {
      "@/*": [
        "src/*"
      ]
    },
    "lib": [
      "esnext",
      "dom",
      "dom.iterable",
      "scripthost"
    ]
  },
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "tests/**/*.ts",
    "tests/**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

**Comment by sleewoo** [comment]
there are two issues:
- `import { Vue } from 'vue-property-decorator';`
- `render(): JSX.Element;`

i do not use `vue-property-decorator` nor `JSX`.
Seems `@vue/cli` installs/configure them by default.
But not everyone uses `@vue/cli`

**Comment by darylteo** [comment]
In vue sfc

`
import vueSlider from 'vue-slider-component';
`

Used directly in vue component.

We are not using vue/cli as it was a legacy app. This is our tsconfig (possibly a very shoddy configuration):

```json
{
    "compilerOptions": {
        "target": "es5",
        "module": "esnext",
        "moduleResolution": "node",
        "allowJs": true,
        "experimentalDecorators": true,
        "strict": true,
        "alwaysStrict": false,
        "strictPropertyInitialization": false,
        "sourceMap": true,
        "typeRoots": [
            "./node_modules/@types"
        ],
        "baseUrl": ".",
        "paths": {
            "src/*": ["src/*"]
        },
        "outDir": ".build",
        "lib": [
            "dom",
            "es6"
        ]
    },
    "include": [
      "src/**/*.ts",
      "src/**/*.vue"
    ],
    "exclude": [
        "node_modules"
    ]
}
```

Installed Dependencies (just updated during our monthly library updates):

```
{
    ...,
    "devDependencies": {
        "@types/chart.js": "^2.7.48",
        "@types/element-resize-detector": "^1.1.2",
        "@types/iscroll": "^5.2.2",
        "@types/jquery": "^3.3.29",
        "@types/lodash": "^4.14.123",
        "@types/moment-timezone": "^0.5.12",
        "@types/node": "^11.11.3",
        "@types/pdfjs-dist": "^2.0.0",
        "autoprefixer": "^9.4.10",
        "cache-loader": "^2.0.1",
        "copy-webpack-plugin": "^5.0.1",
        "css-loader": "^2.1.1",
        "cssnano": "^4.1.10",
        "file-loader": "^3.0.1",
        "fork-ts-checker-webpack-plugin": "^1.0.0",
        "imports-loader": "^0.8.0",
        "mini-css-extract-plugin": "^0.5.0",
        "node-sass": "^4.11.0",
        "optimize-css-assets-webpack-plugin": "^5.0.1",
        "postcss-loader": "^3.0.0",
        "precss": "^4.0.0",
        "resolve-url-loader": "^3.0.1",
        "sass-loader": "^7.1.0",
        "string-replace-webpack-plugin": "^0.1.3",
        "style-loader": "^0.23.1",
        "terser-webpack-plugin": "^1.2.3",
        "thread-loader": "^2.1.2",
        "ts-loader": "^5.3.3",
        "tslint": "^5.14.0",
        "tslint-loader": "^3.6.0",
        "typescript": "^3.3.3333",
        "uglifyjs-webpack-plugin": "^2.1.2",
        "url-loader": "^1.1.2",
        "vue-class-component": "^7.0.1",
        "vue-lazyload": "^1.2.6",
        "vue-loader": "^15.7.0",
        "vue-template-compiler": "^2.6.8",
        "webfonts-loader": "^4.4.0",
        "webpack": "^4.29.6",
        "webpack-cli": "^3.2.3",
        "webpack-dev-server": "^3.2.1",
        "webpack-manifest-plugin": "^2.0.4"
    },
    "dependencies": {
        "babel-polyfill": "^6.26.0",
        "chart.js": "^2.7.3",
        "chartjs-plugin-zoom": "^0.6.6",
        "element-resize-detector": "^1.2.0",
        "es6-promise": "^4.2.6",
        "ics": "^2.13.1",
        "ifvisible": "^1.1.0",
        "joi-browser": "^13.4.0",
        "jquery": "^3.3.1",
        "jquery.nicescroll": "^3.7.6",
        "lodash": "^4.17.11",
        "masonry-layout": "^4.2.2",
        "mobile-detect": "^1.4.3",
        "moment": "^2.24.0",
        "moment-timezone": "^0.5.23",
        "owl.carousel": "^2.3.4",
        "pdfjs-dist": "^2.0.943",
        "raven-js": "^3.27.0",
        "url-polyfill": "^1.1.5",
        "vue": "^2.6.8",
        "vue-clipboard2": "^0.3.0",
        "vue-headful": "^2.0.1",
        "vue-perfect-scrollbar": "^0.1.0",
        "vue-property-decorator": "^8.0.0",
        "vue-router": "^3.0.2",
        "vue-slider-component": "^2.8.14",
        "vuetify": "^1.5.6",
        "vuex": "^3.1.0",
        "vuex-class": "^0.3.1"
    },
    "optionalDependencies": {
        "fsevents": "^2.0.1"
    }
}

```

Hope this helps.

**Comment by NightCatSama** [comment]
The first problem has been discovered, I should install `vue-property-decorator` to `dependencies` instead of `devDependencies`.

The second question I just need to use `/// <reference path="./shims-tsx.d.ts" />` to introduce the `JSX` statement.

But I don't have the latest code at hand, I will update a new version later.

@darylteo @sleewoo  Thanks for feedback.

**Comment by darylteo** [comment]
Can confirm, adding reference to `vue-slider.d.ts` fixes build. 

Will leave open until release. Look forward to the fix and thank you very much for your time and effort. 

D

**Comment by NightCatSama** [comment]
Fixed in version `3.0.13` 😃 .
